google/cloud/storage/_signing.py

```diff
@@ -666,3 +665,33 @@ def _sign_message(message, access_token, service_account_email):
 
     data = json.loads(response.data.decode("utf-8"))
     return data["signature"]
+
+
+def _url_encode(query_params):
+    """Encode query params into URL.
+
+    :type query_params: dict
+    :param query_params: Query params to be encoded.
+
+    :rtype: str
+    :returns: URL encoded query params.
+    """
+    params = []
+    for name, value in query_params.items():
+        params.append(_quote_param(name) + "=" + _quote_param(value))
+
+    return "&".join(sorted(params))
+
+
+def _quote_param(param):
+    """Quote query param.
+
+    :type param: dict
+    :param param: Query param to be encoded.
+
+    :rtype: str
+    :returns: URL encoded query param.
+    """
+    if not isinstance(param, bytes):
+        param = str(param)
+    return six.moves.urllib.parse.quote(param, safe="~")
```

Comment:
Use standard `quote()`  with `~` as safe. This `safe` arg by default has value `/`, so with setting `safe` to `~` we're saying that `/` is not safe.